cranelift/codegen/src/egraph.rs

```diff
@@ -436,14 +436,25 @@ where
             {
                 ScopedEntry::Occupied(o) => {
                     let orig_result = *o.get();
-                    // Hit in GVN map -- reuse value.
-                    self.value_to_opt_value[result] = orig_result;
-                    trace!(" -> merges result {} to {}", result, orig_result);
+                    match (result, orig_result) {
+                        (Some(result), Some(orig_result)) => {
+                            // Hit in GVN map -- reuse value.
+                            self.value_to_opt_value[result] = orig_result;
+                            trace!(" -> merges result {} to {}", result, orig_result);
+                        }
+                        (None, None) => {
+                            trace!(" -> merges with dominating instruction");
+                            // Nothing else to do here.
```

Comment:
"Hit in GVN map but for instruction with zero results: may be a side-effect-only idempotent instruction like `trapnz`/`trapz`. Nothing else to do here." perhaps?